ecr-login/api/factory.go

```diff
@@ -43,6 +44,14 @@ func (defaultClientFactory DefaultClientFactory) NewClient(region string) Client
 	}
 }
 
+// NewClientWithAWSConfig Create new client with AWS Config
```

Comment:
The only Go package I can find right now that imports anything here is [this](https://github.com/concourse/docker-image-resource/blob/4c5d8499b1b850c926b518cb7133294929205981/cmd/check/main.go#L47-L49), which is not calling `NewClient` directly and is performing vendoring.  However, point taken: I would be okay leaving `NewClient` as-is since people may be importing this package.